cdap-common/src/main/java/co/cask/cdap/common/io/Locations.java

```diff
@@ -124,6 +129,64 @@ public long size() throws IOException {
   }
 
   /**
+   * Do some processing on locations contained in the {@code locationDir}, using the {@code processor}.
+   *
+   * @param locationDir location to start iterating from
+   * @param recursive true if this method should be called on the directory {@link Location}s found from
+   *                  {@code locationDir}
+   * @param processor used to process the locations contained by the {@code locationDir}, whether they are directories
+   *                  or not. If the {@link Processor#process} method returns false on any {@link Location} object
+   *                  processed, this method will return the current result of the processor.
+   * @param <R> Type of the return value
+   * @throws IOException if the locations could not be read
+   */
+  public static <R> R processLocations(Location locationDir, boolean recursive, Processor<LocationStatus, R> processor)
```

Comment:
What's the expectation if the given `Location` is not a directory? From the current logic, it's being ignore (i.e. if the given location is a file, it will not get processed).